spec/std/callstack_spec.cr

```diff
@@ -17,9 +17,9 @@ describe "Backtrace" do
     output = `#{tempfile.path}`
 
     # resolved file line:column
-    output.should match(/#{sample} 3:10/) # callee1
-    output.should match(/#{sample} 15:3/) # callee3
-    output.should match(/#{sample} 17:1/) # ???
+    output.should match(/callee1 at #{sample} 3:10/)
+    output.should match(/callee3 at #{sample} 15:3/)
```

Comment:
In osx the `CallStack.decode_function_name(pc)` is not used (ie returns nil).
That is because `@@dwarf_function_names` is empty.

In `self.read_dwarf_sections` the `__debug_info` exists but inside the `Debug::DWARF::Info.new` I think there is something odd as you pointed. Because the `@io.tell < end_offset` in `Debug::DWARF::Info#each` is false from the beginning.

```
info # => Debug::DWARF::Info(
           @abbreviations=nil,
           @address_size=8,
           @debug_abbrev_offset=0,
           @dwarf64=false,
           @io=
            #<File:/path/to/backtrace_sample.dwarf>,
           @offset=566555,
           @ref_offset=0,
           @unit_length=2657,
           @version=4)

{@io.tell, end_offset} # => {731730, 569212}  (this is before the while guard in #each)
```

@ysbaddaden if you can spot/suggest something I am willing to continue.

Otherwise [here](https://github.com/crystal-lang/crystal/files/893815/backtrace.zip) you can find the binary and dwarf I am using to test it.